cmd/end2endtest/lifecycle.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	vapi "go.vocdoni.io/dvote/api"
+	"go.vocdoni.io/dvote/apiclient"
+	"go.vocdoni.io/dvote/log"
+	"go.vocdoni.io/proto/build/go/models"
+)
+
+func init() {
+	ops["lifecyclelection"] = operation{
+		test:        &E2ELifecycleElection{},
+		description: "Publishes two elections one to validate that the election cannot be interrupted and the other validate that the election can be interrupted to valid status",
+		example:     os.Args[0] + " --operation=lifecyclelection --votes=1000",
+	}
+}
+
+var _ VochainTest = (*E2ELifecycleElection)(nil)
+
+type e2eElections []e2eElection
+
+type E2ELifecycleElection struct {
+	e2eElections
+}
+
+func (t *E2ELifecycleElection) Setup(api *apiclient.HTTPclient, c *config) error {
+	t.e2eElections = newE2EElections(api, c, 2)
+
+	electionDescriptions := []struct {
+		d             *vapi.ElectionDescription
+		interruptible bool
+	}{
+		{d: newTestElectionDescription(), interruptible: false},
+		{d: newTestElectionDescription(), interruptible: true},
+	}
+
+	for i, ed := range electionDescriptions {
+		ed.d.ElectionType.Autostart = true
+		ed.d.ElectionType.Interruptible = ed.interruptible
+		ed.d.Census = vapi.CensusTypeDescription{Type: vapi.CensusTypeWeighted}
+
+		if err := t.e2eElections[i].setupElection(ed.d); err != nil {
+			log.Errorf("failed to setup election %d: %v", i, err)
+		}
+	}
+
+	log.Debugf("election details: %+v  %+v", *t.e2eElections[0].election, *t.e2eElections[1].election)
+	return nil
+}
+
+func (t *E2ELifecycleElection) Teardown() error {
+	// nothing to do here
+	return nil
+}
+
+func (t *E2ELifecycleElection) Run() error {
+	var wg sync.WaitGroup
+	errChan := make(chan error, 2)
+
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+
+		api := t.e2eElections[0].api
+		electionID := t.e2eElections[0].election.ElectionID
+
```

Comment:
you could put `isExpectedError := func(err error, status string) bool { ... }` here